keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlArtifactRepository.kt

```diff
@@ -1057,8 +1079,9 @@ class SqlArtifactRepository(
             ArtifactStatus.valueOf(releaseStatus)
           }
           .toSet()
+        // FIXME: in order to sort by properties other than version below, we need the full PublishedArtifact objects
```

Comment:
This is the hardest place to change to use full artifact version objects instead of just version strings that I've come across so far. I think it's safe to say this is the most complex function in all of keel, and I'm scared of breaking it... 😬